### PR TITLE
Add command to delete a network

### LINKED ISSUE #400: Add command to delete a network
It should work like the `vegawallet delete --wallet <WALLET>`, especially the safety mechanism and the `--force` flag.

Example:

```sh
vegawallet network delete --network <NETWORK>
```

### PR DESCRIPTION
**Pull request by pscott31**
Closes #400 
Add a command to delete a network from the store, along with tests for the store, the handler, and an end-to-end test.

### CODE REVIEW

File: cmd/network_delete.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			resp, err := handler(req)
+			if err != nil {
+				return err
+			}
+
+			switch rf.Output {
+			case flags.InteractiveOutput:
+				PrintDeleteNetworkResponse(w, resp)
+			case flags.JSONOutput:
+				return printer.FprintJSON(w, resp)
```
Comment (ValentinTrinque) [inline cmd/network_delete.go:78]: If the deletion is successful, from a script point of view, I think we can just rely on the error code and not print anything for JSON.

File: cmd/network_delete.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		"",
+		"Network to delete",
+	)
+
+	return cmd
+}
+
+type DeleteNetworkHandler func(*network.DeleteNetworkRequest) (*network.DeleteNetworkResponse, error)
+
+func PrintDeleteNetworkResponse(w io.Writer, resp *network.DeleteNetworkResponse) {
```
Comment (ValentinTrinque) [inline cmd/network_delete.go:96]: I think you can rely on the flag to get the Network name, and get rid of the `DeleteNetworkResponse` entirely as it doesn't convey any information we ignore.

File: cmd/network_delete.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	p := printer.NewInteractivePrinter(w)
+	p.NextLine().Text("Network ")
+	p.WarningText(resp.Name)
```
Comment (ValentinTrinque) [inline cmd/network_delete.go:99]: Printing the name is good. I think we should do it in the `delete` command as well (`wallet_delete.go`). Do you mind adding it for consistency ?

File: network/store/v1/store.go [hunk 4]
```diff
@@ -76,6 +76,14 @@ func (s *Store) SaveNetwork(net *network.Network) error {
 	return nil
 }
 
+func (s *Store) DeleteNetwork(name string) error {
+	path := s.GetNetworkPath(name)
+	if err := os.Remove(path); err != nil {
+		return err
+	}
+	return nil
```
Comment (ValentinTrinque) [inline network/store/v1/store.go:84]: if you are not wrapping the error, you can :
```suggestion
	return os.Remove(path)
```

File: tests/sc_delete_network_test.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		"--home", home,
+		"--output", "json",
+		"--network", "my-network-1",
+	})
+
+	// then
+	require.NoError(t, err)
+	require.NotNil(t, deleteNetworkResp)
+	require.Equal(t, "my-network-1", deleteNetworkResp.Name)
+
```
Comment (ValentinTrinque) [inline tests/sc_delete_network_test.go:50]: By removing the response in the JSON output, I think verifying the error would suffice.

**Review by ValentinTrinque — CHANGES_REQUESTED** [review]

File: cmd/network_delete.go (hunk 3 above)
Comment (pscott31) [reply cmd/network_delete.go:99]: Perhaps I'm missing something, but isn't this incompatible with your suggested change above to remove DeleteNetworkResponse?

File: network/store/v1/store.go (hunk 4 above)
Comment (pscott31) [reply network/store/v1/store.go:84]: Thanks, will change

File: cmd/network_delete.go (hunk 3 above)
Comment (pscott31) [reply cmd/network_delete.go:99]: I figured out what you mean here, I can easily get it from the flags. Will add to wallet_delete too

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**